### PR TITLE
feat: shopify-cart in nuxt starter

### PR DESCRIPTION
**Pull request by KrisQ**
### WHY are these changes introduced?

Fixes [[ENG-7070](https://nacelle.atlassian.net/browse/ENG-7070)] 

### WHAT is this pull request doing?

Adds shopify cart to the nuxt starter
Replace checkout with shopify cart checkout

### How to Test

- `git checkout ENG-7070--nuxt-starter-shopify-cart`
- to the nuxt reference store project `.env` add the following: (reach out if you want the vars I used)

```
SHOPIFY_SHOP_ID="<YOUR_SHOPIFY_ID>"
SHOPIFY_STOREFRONT_ACCESS_TOKEN="<YOUR_SHOPIFY_STOREFRONT_ACCESS_TOKEN>"
```

- `cd starter/nuxt`
- `npm i && npm run generate && npm run start`
- see if cart/checkout works

### CODE REVIEW

File: starters/nuxt/pages/index.vue [hunk 1]
```diff
@@ -16,7 +16,10 @@
 
 <script>
 export default {
-  name: 'HomePage'
+  name: 'HomePage',
+  created() {
+    console.log('HELLO2', this.$cartClient);
```
Comment (alexcasche) [inline starters/nuxt/pages/index.vue:21]: Can we remove this?

File: starters/nuxt/pages/products/_handle.vue [hunk 2]
```diff
@@ -121,14 +121,15 @@ export default {
       this.selectedVariant = variant ? { ...variant } : null;
     },
     handleAddItem() {
+      console.log('A', this.quantity);
```
Comment (alexcasche) [inline starters/nuxt/pages/products/_handle.vue:124]: Remove console log

**Review by alexcasche — CHANGES_REQUESTED** [review]
Just some initial console logs to cleanup.  About to do actual testing locally

**Comment by alexcasche** [comment]
Seeing a bug when trying to increment the quantity of a line item.  See attached.  _Could_ be unrelated to your PR but wanted to flag.  It is saying that the merchandiseID is not valid in Shopify.  But for incrementing, no merchandiseId should be necessary

https://user-images.githubusercontent.com/24395931/191546218-10c7d743-54f0-4c62-abb4-ecac1d0a5782.mov

Screen Shot 2022-09-21 at 11 27 23 AM

**Comment by KrisQ** [comment]
> Seeing a bug when trying to increment the quantity of a line item. See attached. _Could_ be unrelated to your PR but wanted to flag. It is saying that the merchandiseId is not valid in Shopify. But for incrementing, no merchandiseId should be necessary
> 
>  Screen.Recording.2022-09-21.at.11.18.46.AM.mov 
> Screen Shot 2022-09-21 at 11 27 23 AM
> 
> Think you just need to remove the `nacelleEntryId` from this action Screen Shot 2022-09-21 at 11 32 29 AM

Hey @alexcasche isn't this a bug in the package that will be resolved once we find a solution for `nacelleEntryId`, it works locally for me but I think I have a temporary workaround in my build for that bug in `shopify-cart`

File: starters/nuxt/pages/cart.vue [hunk 3]
```diff
@@ -5,39 +5,43 @@
       <ul class="cart__items">
         <li
           v-for="lineItem in cart.lineItems"
-          :key="lineItem.id"
+          :key="lineItem.variantId"
           class="cart__item"
         >
+          <pre>{{ JSON.stringify(lineItem, null, 2) }}</pre>
```
Comment (NWRichmond) [inline starters/nuxt/pages/cart.vue:11]: This needs to be removed

```suggestion
```

cart page pre with line item json

File: starters/nuxt/package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "lintfix": "npm run lint:js -- --fix"
   },
   "dependencies": {
+    "@nacelle/shopify-cart": "^0.2.1",
```
Comment (NWRichmond) [inline starters/nuxt/package.json:15]: 1. `@nacelle/shopify-checkout` should be removed.
2. We need the `package-lock.json` updated. Try `npm i --package-lock.json` and bootstrapping again to make sure everything is in sync.

File: starters/nuxt/pages/products/_handle.vue [hunk 5]
```diff
@@ -127,8 +127,8 @@ export default {
       });
       if (variant) {
         this.addItem({
-          variant,
-          quantity: this.quantity
+          ...variant,
+          quantity: parseInt(this.quantity)
```
Comment (NWRichmond) [inline starters/nuxt/pages/products/_handle.vue:131]: Why `parseInt`? Looks like `this.quantity` is already an integer:

product page this quantity number

**Review by NWRichmond — CHANGES_REQUESTED** [review]
This seems mostly complete, but there are some issues.

Other than the issues described in the comments below, the `/cart` experience isn't quite right. When adding an item to the cart from a PDP and navigating to `/cart`, the cart appears empty at first, and doesn't indicate that it's in a loading state:

https://user-images.githubusercontent.com/5732000/191775816-1a611fbf-4e20-4821-80f2-12a907a0b2d9.mov

One way to avoid hard-loading `/cart` in the first place would be to add a
```vue
<nuxt-link to="/cart">View Cart</nuxt-link>
```
to the PDP:

https://user-images.githubusercontent.com/5732000/191776334-dd180b16-7761-410f-93c4-cd654fdc4138.mov

This might be sufficient. If there's a way to simply implement a loading state on the `/cart` page ("Loading Cart..."), then that might be good too, especially when refreshing the `/cart` page.

Lastly, while not related to this PR, let's take this opportunity to use consistent casing in the `buttonText` in `starters/nuxt/pages/products/_handle.vue`. Instead of `'Add To cart'`, let's use `'Add to Cart'`.

**Review by NWRichmond — APPROVED** [review]
Awesome! Thanks so much for those updates, @KrisQ.

https://user-images.githubusercontent.com/5732000/191826850-6ae579b7-2e2b-49ca-9491-f578e91dcec9.mov

**Pull request merged**